frontend/src/metabase/browse/components/CollectionBreadcrumbsWithTooltip.tsx

```diff
@@ -0,0 +1,218 @@
+import styled from "@emotion/styled";
+import classNames from "classnames";
+import type { AnchorHTMLAttributes, MutableRefObject, FC } from "react";
+import { forwardRef, useLayoutEffect, useRef, useState } from "react";
+
+import {
+  ResponsiveChild,
+  ResponsiveContainer,
+} from "metabase/components/ResponsiveContainer/ResponsiveContainer";
+import { useAreAnyTruncated } from "metabase/hooks/use-is-truncated";
+import { color } from "metabase/lib/colors";
+import resizeObserver from "metabase/lib/resize-observer";
+import * as Urls from "metabase/lib/urls";
+import type { AnchorProps, FlexProps } from "metabase/ui";
+import { Anchor, FixedSizeIcon, Flex, Group, Text, Tooltip } from "metabase/ui";
+import type { Collection } from "metabase-types/api";
+
+import { getCollectionName } from "../utils";
+
+export const CollectionBreadcrumbsWithTooltip = ({
+  collection,
+  containerName,
+}: {
+  collection: Collection;
+  containerName: string;
+}) => {
+  const collections = (collection.effective_ancestors || []).concat(collection);
+  const pathString = collections
+    .map(coll => getCollectionName(coll))
+    .join(" / ");
+  const ellipsifyPath = collections.length > 2;
+  const shownCollections = ellipsifyPath
+    ? [collections[0], collections[collections.length - 1]]
+    : collections;
+  const justOneShown = shownCollections.length === 1;
+
+  const { areAnyTruncated, ref } = useAreAnyTruncated<HTMLAnchorElement>();
+
+  const initialEllipsisRef = useRef<HTMLDivElement | null>(null);
+  const [
+    isFirstCollectionDisplayedAsEllipsis,
+    setIsFirstCollectionDisplayedAsEllipsis,
+  ] = useState(false);
+
+  useLayoutEffect(() => {
+    const initialEllipsis = initialEllipsisRef.current;
+    if (!initialEllipsis) {
+      return;
+    }
+    const handleResize = () => {
+      // The initial ellipsis might be hidden via CSS,
+      // so we need to check whether it is displayed via getComputedStyle
+      const style = window.getComputedStyle(initialEllipsis);
+      setIsFirstCollectionDisplayedAsEllipsis(style.display !== "none");
```

Comment:
Not overly sure how I feel about this part haha. On the one hand, using CSS to determine if the width is below the breakpoint seems pretty cool. But it's also not something we see super often. I'd be curious to get a second opinion on this

Reply:
In the interest of time, would it be OK if we leave this as is?